spec/measure_repository_service_test_kit/library_group_spec.rb

```diff
@@ -88,4 +88,389 @@
       expect(result.result).to eq('fail')
     end
   end
+
+  describe 'Server successfully searches and retrieves specified resource by its url' do
+    let(:test) { group.tests[2] }
+    let(:library_url) { 'library_url' }
+
+    it 'passes if the libraries in the returned FHIR searchset bundle match the requested url' do
+      library = FHIR::Library.new(url: library_url)
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?url=#{library_url}"
+      ).to_return(status: 200, body: bundle.to_json)
+
+      result = run(test, url:, library_url:)
+      expect(result.result).to eq('pass')
+    end
+
+    it 'fails if search does not return 200' do
+      library =  FHIR::Library.new(url: library_url)
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?url=#{library_url}"
+      ).to_return(status: 400, body: bundle.to_json)
+
+      result = run(test, url:, library_url:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if the libraries in the returned FHIR searchset bundle do not match the
+      requested url' do
+      library = FHIR::Library.new(url: 'INVALID_URL')
+      bundle = FHIR::Bundle.new(total: 1, entry: [{ resource: library }])
+      stub_request(
+        :get,
+        "#{url}/Library?url=#{library_url}"
+      ).to_return(status: 200, body: bundle.to_json)
+
+      result = run(test, url:, library_url:)
+      expect(result.result).to eq('fail')
+    end
+
+    it 'fails if the search request does not return a FHIR searchset bundle' do
+      library =  FHIR::Library.new(url: 'INVALID_URL')
```

Comment:
```suggestion
      library = FHIR::Library.new(url: 'INVALID_URL')
```

Reply:
Also supernitpick, but I think it's a little misleading to use the `'INVALID_URL'` library here since we're just checking that the output is a bundle. i'd just switch this to the valid url. This goes for all the bundle-check rspec tests :)